lib/ansiblelint/cli.py

```diff
@@ -160,7 +162,7 @@ def get_cli_parser() -> argparse.ArgumentParser:
     return parser
 
 
-def merge_config(file_config, cli_config) -> NamedTuple:
+def merge_config(file_config, cli_config) -> NamedTuple:  # noqa: C901
```

Comment:
I don't have time to do the full review right now but adding complexity is unacceptable.

Reply:
That is not real complexity an you know it well, clearly not a reason to block this change. If you know how to avoid scalability issues with merge_config, you are welcomed to refactor it to avoid triggering C901 when newer option are added.

Reply:
There is a lot of repetition here. It's not complex, but it is redundant. Maybe the complexity warning could be squashed with a refactor. Something along the lines of (untested)

```python
for field in ('quiet', 'parseable', 'parseable_severity' ... ):
    if field in file_config:
        setattr(cli_config, field, getattr(cli_config, field) or file_config[field]))
# Repeat again for addition/extend
```

Reply:
@greg-hellings Yes, it may work but it should be a follow-up and C901 should not be used as a ransom for forcing someone to refactor merge_config. This change itself does not add that kind of complexity we want to avoid with C901.

Reply:
@ssbarnea I agree. Spurious complexity should not be a reason to hold up positive changes. But refactoring should be tossed on the backlog of things to tackle. I've added an issue [here](https://github.com/ansible/ansible-lint/issues/998) for that.

Reply:
While the addition of this branch is not complex in itself, it does underscore this method should be refactored as it grows... I am ok with having that on the backlog as added ^.

Reply:
@awcrosby #999 was already submitted by @ssbarnea to address the complexity refactor.